zipline/data/data_portal.py

```diff
@@ -696,6 +767,26 @@ def get_history_window(self, assets, end_dt, bar_count, frequency, field,
 
         # forward-fill if needed
         if field == "price" and ffill:
+            assets_with_nan_index = np.where(pd.isnull(df.iloc[0, :]))[0]
+            assets_with_leading_nan = assets[assets_with_nan_index]
+
+            if frequency == "1m":
+                data_frequency = 'minute'
+            elif frequency == "1d":
+                data_frequency = 'daily'
+            else:
+                raise Exception(
+                    "Only 1d and 1m are supported for forward-filling.")
+
+            for assets in assets_with_leading_nan:
```

Comment:
interesting, my mental model of how to solve this problem was, when asked for a window on an asset for specific minutes, to extend the window to the last traded date if the first slot of the window has no value.  get the history window (which does all the adjustments), then just chop off all the values before the originally requested window.

your solution seems to work fine, just completely different from what I had mentally envisioned.